src/main/java/chess/domain/piece/PieceType.java

```diff
@@ -0,0 +1,26 @@
+package chess.domain.piece;
+
+import chess.domain.square.Square;
+
+import java.util.function.BiPredicate;
+
+public enum PieceType {
+
+    KING((source, target) -> (source.isStraight(target) || source.isDiagonal(target))
+            && source.isWithinOneStep(target)),
+    QUEEN((source, target) -> source.isStraight(target) || source.isDiagonal(target)),
+    ROOK((source, target) -> source.isStraight(target)),
```

Comment:
```suggestion
    ROOK(Square::isStraight),
```
(반영하지 않으셔도 됩니다.) 개인적으로 호불호가 많이 갈리는 문법이라고 생각하기 때문에 적용하지 않으셔도된다고 생각하는데요. 🤔 혹시나하여 코멘트만 남겨두어요.

Reply:
저도 메서드 참조 방식을 더 선호합니다! 그러나 아래처럼 두 가지 이상의 메서드를 나열할 땐 적용이 안 돼서 통일성 때문에 일단 람다 표현식으로 두었는데, 이 정도까지의 통일성은 안 줘도 될까요?

```java
QUEEN((source, target) -> source.isStraight(target) || source.isDiagonal(target)),
```

Reply:
반영하지 않으셔도 된다고 말씀드렸던 이유가 개인의 취향이라 저는 통일성을 줘도 괜찮고 안줘도 괜찮다고 생각합니다 😄